### PR TITLE
SonarCloud setup

### PR DESCRIPTION
**Pull request by hariso**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,28 +1,72 @@
...
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
 
       - name: Set up JDK 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v1
```
Comment (lovromazgon) [inline .github/workflows/build.yml:18]: How come we downgraded this action?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,28 +1,72 @@
...
+    types: [ opened, synchronize, reopened ]
 jobs:
-  build:
+  tests:
+    name: Tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (lovromazgon) [inline .github/workflows/build.yml:13]: Also this action could be upgraded to v3.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,28 +1,72 @@
...
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
```
Comment (lovromazgon) [inline .github/workflows/build.yml:54]: This action is already at version 3.

**Review by lovromazgon — APPROVED** [review]
Seems like it works, actions are a bit outdated but not a blocker 👍

File: .github/workflows/build.yml (hunk 1 above)
Comment (hariso) [reply .github/workflows/build.yml:18]: Copy-paste.: )

**Pull request merged**